lib/wallet/integration/src/Test/Integration/Scenario/API/Shelley/Transactions.hs

```diff
@@ -2471,15 +2474,15 @@ spec = describe "SHELLEY_TRANSACTIONS" $ do
                 (`shouldSatisfy` (not . null))
             ]
 
-        -- The expectation below is disabled until the ADP-2268 is implemented,
-        -- (The fees aren't guaranteed to be the same between the two endpoints
-        -- because the 'selectCoins' endpoint has already been updated to
-        -- use the new coin selection functionality while the 'postTransaction'
-        -- endpoint still uses the old one, which gives a slightly different
-        -- estimation)
-        -- let apiCoinSelection = getFromResponse Prelude.id coinSelectionResponse
-        -- let fee = computeApiCoinSelectionFee apiCoinSelection
-        -- Quantity (fromIntegral (unCoin (fee))) `shouldBe` expectedFee
+
+        let apiCoinSelection = getFromResponse Prelude.id coinSelectionResponse
+        let fee = computeApiCoinSelectionFee apiCoinSelection
+        let withinToleranceTo (Quantity expected) tol
+                = between (Quantity $ expected - tol, Quantity $ expected + tol)
+        liftIO $ withinToleranceTo
+            expectedFee
+            1_000
+            (Quantity (fromIntegral (unCoin (fee))))
```

Comment:
I seemed to need this to avoid flakiness locally. Not sure why I needed it here, but not on line 2507...

Reply:
Hmm. This is indeed very suspicious.

If we decide to merge this PR without removing the tolerance, then perhaps we can add a TODO with a link to https://input-output.atlassian.net/browse/ADP-2935?

Reply:
> This is indeed very suspicious.

It could be due to slot numbers in the validity interval varying 🤔

Reply:
I've added some comments.